docker/docker_image_dest.go

```diff
@@ -84,7 +84,7 @@ func (d *dockerImageDestination) SupportsSignatures(ctx context.Context) error {
 	case d.c.supportsSignatures:
 		return nil
 	default:
-		return errors.Errorf("X-Registry-Supports-Signatures extension not supported, and lookaside is not configured")
+		return errors.Errorf("X-Registry-Supports-Signatures extension not supported. Configure s/lookaside/sigstore/ or specify a sigstore-staging")
```

Comment:
I’m sorry, the `s/`…`/`…`/` syntax is a jargon for replacing, not a literal suggestion ( http://www.catb.org/~esr/jargon/html/writing-style.html ).

Maybe
```suggestion
		return errors.Errorf("X-Registry-Supports-Signatures extension not supported. Configure a sigstore-staging path")
```
and perhaps point at the relevant man page in to say how; OTOH that may get way too long.

Reply:
I'm not sure of the proper way to link a page etc. but this is the doc I'd link: https://github.com/containers/image/blob/master/docs/containers-registries.d.5.md

Reply:
Well, just include an URL in the page.

OTOH, in the meantime, there is https://github.com/containers/image/pull/1035 , which will make these error messages impossible to hit, so after/if that PR is accepted, this PR can be just closed.

---

@rhatdan (Cc: @QiWang19 ) FYI, one result of #1035 (having built-in defaults for per-user/per-system sigstores) is that copies of signed images to registries that don’t support signatures, and have no sigstore set up, will completely silently succeed, and users will have to figure out that signatures need publishing without much guidance.

That’s always been the case for root (with the configs pointing to `/var/…` for `sigstore-staging`), but now it will be the case for everyone with no way to opt out. I think it’s probably worth it for the simplicity, and anyone who cares about signatures can always configure a default sigstore path. Still, we are adding a silent failure^Wnon-success case for unprivileged users (where before, we were hard-failing unless the user explicitly provided a registries.d configuration that pointed to a writable directory).

Reply:
Should we add an info message when they do not have a separate stage instance, telling that is what is happening?

Reply:
That’s an interesting idea, definitely worth considering.

It would probably be way too noisy when reading signatures (most images are not signed, but we would want to tell the user about the default path exactly when no signatures are found); a reminder when writing signatures (i.e. in `PutSignatures`, when we already know there are signatures to write) could be rather helpful — even if the user knows about the configuration, just to say that the staging data must be published.